### PR TITLE
Update rubygems packages

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update rubygem packages for AZl 3.0

###### Change Log  

- Change: update rubygem-parallel to 1.23.0
- Change: update rubygem-public_suffix to 5.0.3
- Change: update rubygem-multipart-post to 2.3.0
- Change: update rubygem-rdiscount to 2.2.7.1

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Test Methodology

- Pipeline build id: [535408](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=535408&view=results)

### CODE REVIEW

File: SPECS/rubygem-multipart-post/rubygem-multipart-post.spec [hunk 1]
```diff
@@ -30,10 +30,13 @@ gem install -V --local --force --install-dir %{buildroot}/%{gemdir} %{gem_name}-
 
 %files
 %defattr(-,root,root,-)
-%license %{gemdir}/gems/%{gem_name}-%{version}/LICENSE
+%license license.md
 %{gemdir}
 
 %changelog
+* Sat Mar 23 2024 Riken Maharjan <rmaharjan@microsoft.com> - 2.3.0-1
+- Upgrade to 2.3.0 - none
```
Comment (anphel31) [inline SPECS/rubygem-multipart-post/rubygem-multipart-post.spec:38]: what does "none" mean here?

File: SPECS/rubygem-parallel/rubygem-parallel.signatures.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
- "Signatures": {
-  "parallel-1.20.1.tar.gz": "fcc3c400a4a7cc38d7079592437d676160fa9b5e11d0e7224e622c7bbf110351"
- }
-}
+    "Signatures": {
```
Comment (anphel31) [inline SPECS/rubygem-parallel/rubygem-parallel.signatures.json:2]: Can you revert the extra spaces, and make sure your editor keeps a newline at the end of files?

Comment (anphel31) [reply SPECS/rubygem-parallel/rubygem-parallel.signatures.json:2]: In VSCODE the setting is called "Files: Insert Final Newline"

**Review by anphel31 — APPROVED** [review]

**Review by mbykhovtsev-ms — APPROVED** [review]
Approved, but need to address Andrew's comments

File: SPECS/rubygem-multipart-post/rubygem-multipart-post.spec (hunk 1 above)
Comment (rikenm1) [reply SPECS/rubygem-multipart-post/rubygem-multipart-post.spec]: removed. it became from the auto-upgrade tool.

File: SPECS/rubygem-parallel/rubygem-parallel.signatures.json (hunk 2 above)
Comment (rikenm1) [reply SPECS/rubygem-parallel/rubygem-parallel.signatures.json]: removed spaces and using this Insert Final Newline

**Pull request merged**